v3/src/components/tool-shelf/tiles-list-button.tsx

```diff
@@ -0,0 +1,73 @@
+import React from "react"
+import { observer } from "mobx-react-lite"
+import { Menu, MenuButton, MenuItem, MenuList } from "@chakra-ui/react"
+import OptionsIcon from "../../assets/icons/icon-options.svg"
+import { useDocumentContent } from "../../hooks/use-document-content"
+import { uiState } from "../../models/ui-state"
+import { IFreeTileLayout } from "../../models/document/free-tile-row"
+import { t } from "../../utilities/translation/translate"
+import { kRightButtonBackground, ToolShelfButtonTag } from "./tool-shelf-button"
+import GraphIcon from "../../assets/icons/icon-graph.svg"
+import TableIcon from "../../assets/icons/icon-table.svg"
+import SliderIcon from "../../assets/icons/icon-slider.svg"
+import MapIcon from "../../assets/icons/icon-map.svg"
+import CalculatorIcon from "../../assets/icons/icon-calc.svg"
+import GuideIcon from "../../assets/icons/icon-guide.svg"
+import WebViewIcon from "../../assets/icons/icon-media-tool.svg"
+
+import "./tool-shelf.scss"
+
+export const TilesListShelfButton = observer(function TilesListShelfButton() {
+  const documentContent = useDocumentContent()
+  const tilesArr: any[] = []
+
+  documentContent?.tileMap.forEach((tile) => {
+    tilesArr.push(tile)
+  })
```

Comment:
I think something like this should work:
```suggestion
  const tilesArr = documentContent ? values(documentContent.tileMap) : []
```

`values` is imported from `mobx`. I got this from ChatGPT and didn't test it so it might be wrong, but I'm sure there's some cleaner way of doing this.